packages/prover/src/proof_provider/proof_provider.ts

```diff
@@ -0,0 +1,182 @@
+      genesisData,
+    });
+
+    assertLightClient(this.lightClient);
+    // Wait for the lightclient to start
+    await new Promise<void>((resolve) => {
+      const lightClientStarted = (status: RunStatusCode): void => {
+        if (status === RunStatusCode.started) {
+          this.lightClient?.emitter.off(LightclientEvent.statusChange, lightClientStarted);
+          resolve();
+        }
+      };
+      this.lightClient?.emitter.on(LightclientEvent.statusChange, lightClientStarted);
+      this.lightClient?.start();
+    });
+    this.registerEvents();
+
+    // Load the payloads from the CL
+    const {start, end} = await getUnFinalizedRangeForPayloads(this.lightClient);
+    const payloads = await getExecutionPayloads(this.opts.api, start, end);
+    for (const payload of Object.values(payloads)) {
+      this.store.set(payload, false);
+    }
+
+    // Load the finalized payload from the CL
+    const finalizedSlot = this.lightClient.getFinalized().beacon.slot;
+    const finalizedPayload = await getExecutionPayloads(this.opts.api, finalizedSlot, finalizedSlot);
+    this.store.set(finalizedPayload[finalizedSlot], true);
+  }
+
+  getStatus(): {latest: number; finalized: number; status: RunStatusCode} {
+    if (!this.lightClient) {
+      return {
+        latest: 0,
+        finalized: 0,
+        status: RunStatusCode.uninitialized,
+      };
+    }
+
+    return {
+      latest: this.lightClient.getHead().beacon.slot,
+      finalized: this.lightClient.getFinalized().beacon.slot,
+      status: this.lightClient.status,
+    };
+  }
+
+  async getExecutionPayload(blockNumber: number | string | "finalized" | "latest"): Promise<allForks.ExecutionPayload> {
+    assertLightClient(this.lightClient);
+
+    if (typeof blockNumber === "string" && blockNumber === "finalized") {
+      const payload = this.store.finalized;
+      if (!payload) throw new Error("No finalized payload");
+      return payload;
+    }
+
+    if (typeof blockNumber === "string" && blockNumber === "latest") {
+      const payload = this.store.latest;
+      if (!payload) throw new Error("No latest payload");
+      return payload;
+    }
+
+    if ((typeof blockNumber === "string" && blockNumber.startsWith("0x")) || typeof blockNumber === "number") {
+      const payload = await this.store.get(blockNumber);
+      if (!payload) throw new Error(`No payload for blockNumber ${blockNumber}`);
+      return payload;
+    }
+
+    throw new Error(`Invalid blockNumber "${blockNumber}"`);
+  }
+
+  async processLCHeader(lcHeader: allForks.LightClientHeader, finalized = false): Promise<void> {
+    const fork = this.opts.config.getForkName(lcHeader.beacon.slot);
+
+    if (!isForkExecution(fork)) {
+      return;
+    }
+
+    if (
+      isForkExecution(fork) &&
+      (!("execution" in lcHeader) || lcHeader.execution === ssz.capella.ExecutionPayloadHeader.defaultValue())
+    ) {
+      throw new Error("Execution payload is required for execution fork");
```

Comment:
you should check for isForkWithdrawals or something here as only they will have a non zero execution, also you can't === objects you will need to use : `ssz.capella.ExecutionPayloadHeader.equals(lcHeader.execution,ssz.capella.ExecutionPayloadHeader.defaultValue())

also the ssz type for above will depend upon the fork, so you will actually have to do:

`const sszType =config.getForkExecutionTypes(lcHeader.beacon.slot).ExecutionPayloadHeader`
and then check `sszType.equals(lcHeader.execution,sszType.defaultValue())`

Reply:
Updated the `sszType` logic. For the `isForkExecution` seems correct to me. If that's not correct then we may need to update `isForkExecution`.

Reply:
bellatrix which is execution fork will not have `execution` in LCHeader and should just return, also could you chain this with above confition on L151 with `if else`

Reply:
I don't get your point. If `bellatrix` is an execution fork why would it not have `execution` in the header?

Reply:
execution is only bundled with LightClient header on/after capella

Reply:
@g11tech Updated the code to `isForkWithdrawals`